src/front/c/mod.rs

```diff
@@ -0,0 +1,668 @@
+                    })
+                    .val;
+
+                for _ in 0..bound {
+                    self.circ.enter_scope();
+                    self.gen_stmt(for_stmt.node.statement.node.clone());
+                    self.circ.exit_scope();
+                    self.gen_expr(for_stmt.node.step.as_ref().unwrap().node.clone());
+                }
+                self.circ.exit_scope();
+            }
+            _ => unimplemented!("Statement {:#?} hasn't been implemented", stmt),
+        }
+    }
+
+    fn gen(&mut self) {
+        let TranslationUnit(nodes) = self.tu.clone();
+        for n in nodes.iter() {
+            match n.node {
+                ExternalDeclaration::Declaration(ref decl) => {
+                    debug!("{:#?}", decl);
+                }
+                ExternalDeclaration::FunctionDefinition(ref fn_def) => {
+                    debug!("{:#?}", fn_def.node.clone());
+                    // println!("function: {:#?}", fn_def.node);
+                    let fn_info = ast_utils::get_fn_info(&fn_def.node);
+                    self.circ
+                        .enter_fn(fn_info.name.to_owned(), fn_info.ret_ty.clone());
+                    for arg in fn_info.args.iter() {
+                        // TODO: self.gen_decl(arg);
+                        let p = &arg.specifiers[0];
+                        let vis = self.interpret_visibility(&p.node);
+                        let base_ty = d_type_(arg.specifiers[1..].to_vec());
+                        let d = &arg.declarator.as_ref().unwrap().node;
+                        let derived_ty = self.derived_type_(base_ty.unwrap(), d.derived.to_vec());
+                        let name = name_from_decl(d);
+                        let res = self.circ.declare(name.clone(), &derived_ty, true, vis);
+                        self.unwrap(res);
+                    }
+                    self.gen_stmt(fn_info.body.clone());
+                    if let Some(r) = self.circ.exit_fn() {
+                        match self.mode {
+                            Mode::Mpc(_) => {
+                                let ret_term = r.unwrap_term();
+                                let ret_terms = ret_term.term.terms();
+                                self.circ
+                                    .cir_ctx()
+                                    .cs
+                                    .borrow_mut()
+                                    .outputs
+                                    .extend(ret_terms);
+                            }
+                            Mode::Proof => {
+                                let ty = fn_info.ret_ty.as_ref().unwrap();
+                                let name = "return".to_owned();
+                                let term = r.unwrap_term();
+                                let _r = self.circ.declare(name.clone(), ty, false, PROVER_VIS);
+                                self.circ.assign_with_assertions(name, term, ty, PUBLIC_VIS);
```

Comment:
I would actually do `unimplemented!()` here, unless you're going to test this support.